src/Std/Tactic/BVDecide/Normalize/BitVec.lean

```diff
@@ -257,6 +257,14 @@ theorem BitVec.add_const_right' (a b c : BitVec w) : (a + b) + c = (b + c) + a :
 attribute [bv_normalize] BitVec.mul_zero
 attribute [bv_normalize] BitVec.zero_mul
 
+@[bv_normalize]
+theorem BitVec.neg_mul' (x y : BitVec w) : (~~~x + 1#w) * y = ~~~(x * y) + 1#w := by
```

Comment:
As a general principle, unless there is a clear local convention, I would like all theorems with primes in their names to have a doc-comment explaining the naming convention / name clash / difficulty in naming.

Reply:
@hargoniX can comment here as to his convention for the bv_normalize code, as I think I'm following his local convention. (The lemma is specifically applied to the normalized form where negation has been expanded to twos complement, and isn't 'user-facing' in that sense.)

Reply:
These lemmas should be named `neg_mul` (note that they exist in their own namespace separate from regular `BitVec` theory so you can reuse the name). The reason that there is `'` lemmas above is that you also need to match on `(1#w + ~~~x) * y = ~~~(x * y) + 1#w`. This is what the symmetries remark in the spreadsheet is alluding to. If you read the Bitwuzla code you will see they have a function `is_bv_neg` https://github.com/bitwuzla/bitwuzla/blob/main/src/rewrite/rewriter.cpp#L293 that checks for all these symmetries.

Reply:
Ah, this makes sense, thanks.